### PR TITLE
add "command" annotation for executing after secret template is rendered

### PR DESCRIPTION
**Pull request by joemiller**
Addresses #56 

This PR adds a new annotation:
* `vault.hashicorp.com/agent-inject-command-` 

This annotation similar to the one used for customizing the secret template. The specified command will be run by vault-agent after the secret is rendered.

Unaddressed by this PR is that in many cases the vault agent will be running as a different user than the main app container. Thus, the example use case from #56 is not yet fully possible since the call to `pkill` will be run as a different user/uid than the nginx container.

It may be desirable to allow an annotation that will set the UID of the vault agent container to address this.

**EDIT**: support for specifying the userid/groupid of the vault agent sidecar is addressed in #60

### CODE REVIEW

**Comment by jasonodonnell** [comment]
Hi @joemiller, my apologies for the delay.  I'm ready to review this PR.  Can you resolve the conflicts in the meantime? Thanks!

**Comment by joemiller** [comment]
Will do. I’m traveling so it may be a few days

**Comment by joemiller** [comment]
@jasonodonnell conflicts resolved. The conflict in annotations_test.go was a bit of a challenge to resolve but I believe it's in the correct state now. Please advise if I missed something there. thanks!

File: agent-inject/agent/config.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ type Template struct {
 	Contents       string `json:"contents"`
 	LeftDelim      string `json:"left_delimiter,omitempty"`
 	RightDelim     string `json:"right_delimiter,omitempty"`
+	Command        string `json:"command",omitempty`
```
Comment (jasonodonnell) [inline agent-inject/agent/config.go]: The closing quote needs to be at the end of the line before the tick.

File: agent-inject/agent/annotations.go [hunk 2]
```diff
@@ -38,6 +38,12 @@ const (
 	// AnnotationAgentInjectToken is the annotation key for injecting the token
 	// from auth/token/lookup-self
 	AnnotationAgentInjectToken = "vault.hashicorp.com/agent-inject-token"
+	// AnnotationAgentInjectCommand is the key annotation that configures Vault Agent
```
Comment (jasonodonnell) [inline agent-inject/agent/annotations.go]: Nit: can you add a newline before this comment?

**Comment by joemiller** [comment]
@jasonodonnell great eye! thanks. Fixed, pushed

**Comment by sbeaulie** [comment]
Are you planning to get a release for this soon?

**Comment by Shaeli** [comment]
Hello,
Any news on when this will be released? 
Thanks

**Comment by tvoran** [comment]
@Shaeli @sbeaulie It should go out this week, in #99

**Comment by joemiller** [comment]
It would be great to get it released together with the run-as-user/group
support in https://github.com/hashicorp/vault-k8s/pull/60 since your
command may need to be run as a user/group other than the vault defaults
(100/1000) but that PR is still WIP (hopefully done soon)

**Pull request merged**